### PR TITLE
AMQP-597: Document MetaAnnotations

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/AMQP-597

Also clean up extra queues left due to `@DirtiesContext` on some tests.

### CODE REVIEW

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java [hunk 1]
```diff
@@ -1061,6 +1080,30 @@ public String baz(Baz baz, String rk) {
 
 	}
 
+	@Target({ElementType.TYPE, ElementType.METHOD, ElementType.ANNOTATION_TYPE})
+	@Retention(RetentionPolicy.RUNTIME)
+	@RabbitListener(bindings = @QueueBinding(
+			value = @Queue,
+			exchange = @Exchange(value = "test.metaFanout", type = ExchangeTypes.FANOUT, autoDelete = "true")))
+	public @interface MyAnonFanoutListener {
```
Comment (artembilan) [inline spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java]: I wonder if they works well as really full meta-annotation.
I mean how about the `value` attribute in the custom annotation, which is visible from the annotation processor.
Do we use here in the SA the same mechanism as we do in the SI ?
Thanks

Comment (artembilan) [inline spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java]: M-m-m. Looks like no. We do only this:
```
private Collection<RabbitListener> findListenerAnnotations(Method method) {
```
but don't collect attributes or `AnnotationUtils.synthesizeAnnotation()` from the custom environment. 

OK. That's fine too.

More complex cases should be achieved with the regular `@Bean` definition for the AMQP entities (exchanges, queues and bidnings between them) and used as a `queus` from the normal `@RabbitListener` annotation.

How about to document that as well?

**Comment by garyrussell** [comment]
Pushed

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/bfc854b42c1ef1c28d9444b03af68b5aac90734d after rebase.

**Pull request closed without merging**